### PR TITLE
Hit the API

### PR DESCRIPTION
**Pull request by AlexHamn**
### Changes in this branch

- Create a new game using the API.
- Implement the "Refresh" button (receiving data from the API and parsing the JSON).
- Implement the form "Submit" button (sending data to the API).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,57 @@
...
+
+const scores = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/7oyqKQQCxsTaogFAwN0N/scores/';
+
+let scoresArray = [];
+
+const scoresList = document.getElementById('scoresList');
+
+const refreshButton = document.getElementById('refresh');
+
+function displayScores() {
```
Comment (rukundoeric) [inline src/index.js]: - [x] <b>REQUIRED:<b/> Prefer using ES6 arrow function. Actually, there is nothing wrong with ES5 functions syntax but it is mandatory to use es6 features in the javascript module. besides, Arrow functions provide a simplified structure, allowing for a more concise way to write function expressions. which makes them better than ES5 functions(function declarations). Here is the syntax of an ES6 arrow function...
```javascript
  const functionName = () => {
     // code here
  }
```

File: src/index.js [hunk 2]
```diff
@@ -1 +1,57 @@
...
+}
+
+document.addEventListener('DOMContentLoaded', displayScores);
+refreshButton.addEventListener('click', displayScores);
+
+const addButton = document.getElementById('addButton');
+const user = document.getElementById('user');
+const score = document.getElementById('score');
+
+function addScore(e) {
```
Comment (rukundoeric) [inline src/index.js:41]: - [x] <b>REQUIRED:<b/> Prefer using ES6 arrow function. Actually, there is nothing wrong with ES5 functions syntax but it is mandatory to use es6 features in the javascript module. besides, Arrow functions provide a simplified structure, allowing for a more concise way to write function expressions. which makes them better than ES5 functions(function declarations). Here is the syntax of an ES6 arrow function...
```javascript
  const functionName = () => {
     // code here
  }
```

File: src/index.js [hunk 3]
```diff
@@ -1 +1,57 @@
...
+  e.preventDefault();
+
+  // if (user.value && typeof score.value === 'number') {
```
Comment (rukundoeric) [inline src/index.js]: - [x] <b>REQUIRED:</b> To make sure that your codebase is always clean, remember to remove commented-codes.

File: src/index.js [hunk 4]
```diff
@@ -1 +1,57 @@
...
+  fetch(scores, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  })
+    .then((res) => res.json())
+    .then((data) => {
+      scoresArray = data.result;
+    });
```
Comment (rukundoeric) [inline src/index.js]: - [x] <b>REQUIRED:</b> Kindly, use `async/await` here, to comply with project specifications.

File: src/index.js [hunk 5]
```diff
@@ -1 +1,57 @@
...
+
+  scoresArray.forEach((score) => {
+    const li = document.createElement('li');
+    li.innerHTML = `<p>${score.user}: ${score.score}</p>`;
+    scoresList.append(li);
+  });
+}
+
+document.addEventListener('DOMContentLoaded', displayScores);
+refreshButton.addEventListener('click', displayScores);
```
Comment (rukundoeric) [inline src/index.js:35]: - [x] <b>REQUIRED:</b> Refresh button is working only when it is clicked twice. kindly fix that.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @AlexHamn, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  
## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by oxenprogrammer — APPROVED** [review]
Hi @AlexHamn , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**